src/wasm/wasm-type.cpp

```diff
@@ -289,6 +299,16 @@ Type getType(unsigned size, bool float_) {
   WASM_UNREACHABLE("invalid size");
 }
 
+bool isLeftSubTypeOfRight(Type left, Type right) {
```

Comment:
You mean something like this?
```
return getLeastUpperBound(left, right) == right;
```

I don't think I can do this, because `getLeastUpperBound` does special handling for `none` and `unreachable` types. For example, when given `unreachable` and `i32`, `getLeastUpperBound` will return `i32`, but this doesn't mean `unreachable` is a subtype of `i32`. And again, `getLeastUpperBound` returns `none` when given `none` and any other type, but this does not mean those types are subtype of `none`.

The reason `getLeastUpperBound` includes that special handling for `none` and `unreachable` is, this is the way it works when we 'merge' types of different branches targeting a single block. This routine was scattered throughout the code base and now it is done in this single place.